lib/Config.js

```diff
@@ -67,13 +88,83 @@ class Config {
             });
         }
 
-        assert(typeof config.regions === 'object',
-               'bad config: the list of regions is mandatory');
-        assert(Object.keys(config.regions).every(
-               r => typeof r === 'string' && config.regions[r] instanceof Array
-               && config.regions[r].every(e => typeof e === 'string')),
-               'bad config: regions must be a set of {region: [endpoints]}');
-        this.regions = config.regions;
+        // legacy
+        if (config.regions !== undefined) {
+            assert(config.locationConstraints === undefined &&
+                config.restEndpoints === undefined,
+                'bad config: use either regions or locationConstraints/' +
+                'restEndpoints configuration, not both');
+            assert(typeof config.regions === 'object',
+                'bad config: regions must be an object');
+            assert(Object.keys(config.regions).every(
+                r => typeof r === 'string' && config.regions[r] instanceof Array
+                && config.regions[r].every(e => typeof e === 'string')),
+                'bad config: regions must be a set of {region: [endpoints]}');
+            this.regions = config.regions;
+        }
+
+        if (config.locationConstraints !== undefined) {
```

Comment:
please add tests for this logic.  if it is easier to abstract into a function in order to test, do that.